yeat/workflows/workflows.py

```diff
@@ -7,39 +7,38 @@
 # Development Center.
 # -------------------------------------------------------------------------------------------------
 
-from pathlib import Path
+from . import bandage
 from pkg_resources import resource_filename
 from snakemake import snakemake
 
 
+PAIRED = ["spades", "megahit", "unicycler"]
+PACBIO = ["canu", "flye"]
+
+
 def run_paired(
-    fastq1,
-    fastq2,
+    assembly_samples,
     assembly_configs,
     outdir=".",
     cores=1,
-    sample="sample",
     dryrun="dry",
-    downsample=0,
     coverage=150,
-    seed=None,
+    downsample=0,
     genomesize=0,
+    seed=None,
 ):
     snakefile = resource_filename("yeat", "workflows/snakefiles/Paired")
-    r1 = Path(fastq1).resolve()
-    if not r1.is_file():
-        raise FileNotFoundError(f"No such file: '{r1}'")
-    r2 = Path(fastq2).resolve()
-    if not r2.is_file():
-        raise FileNotFoundError(f"No such file: '{r2}'")
-    assemblers = [config.algorithm for config in assembly_configs]
-    extra_args = {config.algorithm: config.extra_args for config in assembly_configs}
+    samples = {k: v.sample for k, v in assembly_samples.items()}
+    labels = [config.label for config in assembly_configs]
+    assemblers = {config.label: config.algorithm for config in assembly_configs}
+    extra_args = {config.label: config.extra_args for config in assembly_configs}
+    label_to_samples = {config.label: config.samples for config in assembly_configs}
```

Comment:
Seems like there's a lot of redundancy between this `run_paired` code and code in `run_pacbio`. That's a maintenance risk!

Reply:
Created the `to_dict()` in the `AssemblerConfig` object. Returns back a nested dictionary that each workflow uses: Paired, Pacbio, and Bandage.